raven/utilities/forecasting.py

```diff
@@ -173,3 +182,98 @@ def perform_climatology_esp(model_name, forecast_date, forecast_duration, **kwds
     qsims["member"] = (["member"], avail_years)
 
     return qsims
+
+
+def get_recent_ECCC_forecast(region, climate_model='GEPS'):
```

Comment:
This function should be split in two: 
1. Get netCDF resource from PAVICS THREDDS, run health checks. 
2. Compute region average. 

Step 2 could be refactored in a couple of months to use  pangeo-netCDF. This would require CF coordinate bounds are included in the variables. An advantage is that averaging weights can be stored and reloaded to speed things up.

Reply:
Step 2 would be implemented into clisops, on which xclim relies for subsetting and regridding operations.

Reply:
I have refactored into two functions.